src/components/popups/TeamSetting.tsx

```diff
@@ -0,0 +1,195 @@
+import React, { useState, useEffect } from "react";
+import "../../styles/popups/TeamSetting.scss";
+import { api, handleError } from "helpers/api";
+import PropTypes from "prop-types";
+import { Button } from "components/ui/Button";
+import { useNavigate, useParams } from "react-router-dom";
+
+const TeamSettings = ({ isOpen, onClose }) => {
+  //const [editMode, setEditMode] = useState(false); EDIT Mode: for extensibility in M4
+  const [teamName, setTeamName] = useState();
+  const [teamDescription, setTeamDescription] = useState();
+  const [teamUUID, setTeamUUID] = useState();
+  const [teamMembers, setTeamMembers] = useState([]);
+  const [inviteURL, setInviteURL] = useState();
+  const [copied, setCopied] = useState("");
+  const { teamId } = useParams();
+  const token = localStorage.getItem("token");
+  const userId = localStorage.getItem("id");
+  const [error, setError] = useState("");
+  const [leaveError, setLeaveError] = useState("");
+  const baseURL = window.location.origin;
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    const fetchUserTeam = async () => {
+      try {
+        const response = await api.get(`/api/v1/users/${userId}/teams`, {
+          headers: { Authorization: `${token}` },
+        });
+        const foundTeam = response.data.find(
+          (team) => team.teamId === parseInt(teamId)
+        );
+        setTeamName(foundTeam.name);
+        setTeamDescription(foundTeam.description);
+        setTeamUUID(foundTeam.teamUUID);
+        setInviteURL(`${baseURL}/invitation/${foundTeam.teamUUID}`);
+      } catch (error) {
+        setError("Something went wrong. Please try again");
+        if (error.response.status === 401) {
```

Comment:
@Monti934 @B1s9l , if 404 and/or 401, is then the user rerouted to their team overview?

Reply:
- [ ] will think about this later, at the moment the user would not see the error message anymore
- [ ] maybe add a button with error to go back